src/discount/Product.java

```diff
@@ -0,0 +1,39 @@
+package discount;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class Product {
+
+    private String type;
+    private double price;
+    private boolean discount;
+
+    public Product(String type, double price, boolean discount) {
+        this.type = type;
+        this.price = price;
+        this.discount = discount;
+    }
+
+    public String getType() {
+        return type;
+    }
+
+    public double getPrice() {
+        return price;
+    }
+
+    public Product getCheapestBook(List<Product> products) throws Exception {
```

Comment:
Sorry, but this is wrong. You should find all the products of type book (you did it correctly), check whether your product has a discount, and if it has, then subtract 10% from the price the product has and set it. Check the task description for more details (2.2)